### PR TITLE
invalidate tag of content and node when update used media

### PR DESCRIPTION
**Pull request by alavieille**
[OO-FEATURE] Invalidate the cache tag of contents and nodes when update a media which they use it

https://github.com/open-orchestra/open-orchestra-model-bundle/pull/632
https://github.com/open-orchestra/open-orchestra-model-interface/pull/216
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/277

### CODE REVIEW

File: MediaAdminBundle/ExtractReference/ExtractReferenceInterface.php [hunk 1]
```diff
@@ -33,6 +40,15 @@ public function extractReference(StatusableInterface $statusableElement);
     public function getReferencePattern($statusableElementId);
 
     /**
+     * Get invalidate tag of statusable element for reference
+     *
+     * @param string $reference
+     *
+     * @return string|null
+     */
+    public function getInvalidateTagStatusableElement($reference);
```
Comment (ngilain) [inline MediaAdminBundle/ExtractReference/ExtractReferenceInterface.php]: @alavieille : the purpose of this method is not clear, could you rename it into something more relevant ? What is an invalidate tag, is it simply a cache tag ?
And not sure you should mention StatusableElement in the naming as it is not related to the parameters

File: MediaAdminBundle/ExtractReference/ExtractReferenceManager.php [hunk 2]
```diff
@@ -59,4 +59,29 @@ public function getReferencePattern(StatusableInterface $statusableElement)
...
+                if ($strategy->supportReference($reference)) {
+                    $tag = $strategy->getInvalidateTagStatusableElement($reference);
+                    if (null !== $tag) {
+                        $invalidateTag[] = $tag;
+                    }
+                }
+            }
+        }
+
+        return $invalidateTag;
```
Comment (ngilain) [inline MediaAdminBundle/ExtractReference/ExtractReferenceManager.php]: @alavieille : an array is return but the name of the method and the phpdoc on the interface both say only a string should be returned

**Comment by alavieille** [comment]
@ngilain pr updated

**Comment by ngilain** [comment]
:+1:

**Pull request merged**